src/hotspot/share/opto/cfgnode.cpp

```diff
@@ -2311,6 +2311,15 @@ Node *PhiNode::Ideal(PhaseGVN *phase, bool can_reshape) {
             }
           }
         }
+
+        if (igvn) {
+          // Already replace this phi node to cut it off from the graph to not interfere in dead loop checks during the
+          // transformations of the new phi nodes below. Otherwise, we could wrongly conclude that there is no dead loop
+          // because we are finding this phi node again. Also set the type of the new MergeMem node in case we are also
+          // visiting it in the transformations below.
+          igvn->replace_node(this, result);
+          igvn->set_type(result, result->bottom_type());
```

Comment:
Did you consider simply cut off Phi's inputs with `top` by using `replace_input_of()`?

Reply:
Unfortunately, this does not work. When doing the dead loop check for `11853 Phi` in the `phase->transform(phi)` call below, we find the `this` phi (`989 Phi`) by following the inputs of `11853 Phi` and conclude it's dead loop safe. This means we would need to cut the outputs of `989 Phi`. The only good way I found to achieve that is to directly replace the `this` phi here with the resulting mergemem node. Otherwise, we somehow need to restore the outputs again after the transformations to correctly subsume the `this` phi later.

Reply:
Okay.